.travis.yml

```diff
@@ -0,0 +1,42 @@
+sudo: false
+
+addons:
+  apt:
+    packages:
+      - oracle-java8-installer
+
+language: java
+
+dist: trusty
+
+jdk:
+  - openjdk8
+
+install: true # skips travis' default installation step which executes gradle assemble.
+
+script: ./gradlew clean build
+
+# The before_cache and the cache steps cache the gradle installation on travis.
+before_cache:
```

Comment:
in the same time they make an example where cache is setup as a monolithic block:
https://docs.travis-ci.com/user/languages/java/#Caching